pkg/releaseutil/manifest_sorter.go

```diff
@@ -142,6 +152,25 @@ func (file *manifestFile) sort(result *result) error {
 			continue
 		}
 
+		expectation, ok := entry.Metadata.Annotations["helm.sh/test-expect-success"]
```

Comment:
Is there currently a requirement that tests be defined in the `templates/tests` directory? (that seems to be indicated by the PR/commit comments but I'm missing where it's done in the code).

I think I'd rather only use the annotation. See also https://github.com/helm/community/pull/40/files#r297251345